### PR TITLE
WT-2065 Add a quota to shared cache configuration.

### PR DESCRIPTION
**Pull request by agorrod**
That provides the ability to limit how much cache a single participant in a shared cache can be allocated.

### CODE REVIEW

**Comment by agorrod** [comment]
@michaelcahill Would you mind reviewing change?

File: src/conn/conn_cache_pool.c [hunk 1]
```diff
@@ -156,6 +161,9 @@ __wt_cache_pool_config(WT_SESSION_IMPL *session, const char **cfg)
 		    session, cfg, "shared_cache.chunk", &cval));
 		WT_ASSERT(session, cval.val != 0);
 		chunk = (uint64_t)cval.val;
+		WT_ERR(__wt_config_gets(
+		    session, cfg, "shared_cache.quota", &cval));
+		quota = (uint64_t)cval.val;
 	}
```
Comment (michaelcahill) [inline src/conn/conn_cache_pool.c]: Three is about as many options as I'd like to repeat here -- is it reasonable to create an inlined function that handles both cases and call it three times?

**Comment by michaelcahill** [comment]
@agorrod, LGTM -- we can worry about the config code separately if you prefer.

**Comment by agorrod** [comment]
> we can worry about the config code separately if you prefer

@michaelcahill Yes please. I coded up a macro wrapper and it's not an obvious improvement. I'd rather push it into a different pull request.

**Pull request merged**